### PR TITLE
Bug 1805398: Disable delegated auth for recovery

### PR DESCRIPTION
**Pull request by tnozicka**
Delegated AuthN/AuthZ do not support setting tls SNI name, nor they are needed for certregenerationcontroller to run. The failure to read those is fatal and breaks the certregenerationcontroller before it even starts.

```
W0217 17:11:14.501686       1 cmd.go:195] Using insecure, self-signed certificates
I0217 17:11:14.502018       1 crypto.go:580] Generating new CA for cert-regeneration-controller-signer@1581959474 cert, and key in /tmp/serving-cert-276295724/serving-signer.crt, /tmp/serving-cert-276295724/serving-signer.key
I0217 17:11:15.746589       1 observer_polling.go:155] Starting file observer
W0217 17:11:15.761254       1 builder.go:174] unable to get owner reference (falling back to namespace): Get https://localhost:6443/api/v1/namespaces/openshift-kube-apiserver/pods: x509: certificate has expired or is not yet valid
W0217 17:11:16.483729       1 configmap_cafile_content.go:102] unable to load initial CA bundle for: "client-ca::kube-system::extension-apiserver-authentication::client-ca-file" due to: configmap "extension-apiserver-authentication" not found
...
W0217 17:11:45.494698       1 configmap_cafile_content.go:102] unable to load initial CA bundle for: "client-ca::kube-system::extension-apiserver-authentication::client-ca-file" due to: configmap "extension-apiserver-authentication" not found
W0217 17:11:45.495003       1 configmap_cafile_content.go:102] unable to load initial CA bundle for: "client-ca::kube-system::extension-apiserver-authentication::requestheader-client-ca-file" due to: configmap "extension-apiserver-authentication" not found
F0217 17:11:46.482278       1 cmd.go:120] unable to load configmap based request-header-client-ca-file: Get https://localhost:6443/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication: x509: certificate has expired or is not yet valid
```

Requires:
- [x] https://github.com/openshift/library-go/pull/715

/cc @deads2k @sttts

### CODE REVIEW

File: bindata/v4.1.0/kube-apiserver/certregenerationcontroller-config-cm.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-apiserver-cert-regeneration-controller-config
+  namespace: openshift-kube-apiserver
+data:
+  kubeconfig: |
```
Comment (deads2k) [inline bindata/v4.1.0/kube-apiserver/certregenerationcontroller-config-cm.yaml]: key doesn't match content

File: bindata/v4.1.0/kube-apiserver/certregenerationcontroller-config-cm.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    authentication:
+      disabled: true
+    authorization:
+      disabled: true
```
Comment (deads2k) [inline bindata/v4.1.0/kube-apiserver/certregenerationcontroller-config-cm.yaml]: Allowing unrestricted access to debug and metrics hasn't been vetted as viable and risks escalation bugs.  Can you shut off the entire server in the binary instead until the option can be plumbed?

Comment (tnozicka) [reply bindata/v4.1.0/kube-apiserver/certregenerationcontroller-config-cm.yaml]: ok, disabled it

**Comment by tnozicka** [comment]
```

**Pull request merged**